hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkDataSetProcessor.java

```diff
@@ -343,23 +343,17 @@ public Partitioner getPartitioner(DataSet<ExecRow> dataSet, ExecRow template, in
                                           double sampleFraction) throws StandardException {
         try {
             Dataset<Row> table = null;
+            ExternalTableUtils.preSortColumns(tableSchema.fields(), partitionColumnMap);
 
             try {
-                DataSet<V> empty_ds = checkExistingOrEmpty( location, context );
-                if( empty_ds != null ) return empty_ds;
```

Comment:
So if I understand correctly, you removed the redundant check for directory existence and you move it to exception handling logic during the actual read, right?

Reply:
yes. spark is anyhow checking first if the directory exists. so if the directory would exist, we do this check twice, which is costing us some performance on EVERY read (depending on ping ~300ms). of course we want to be good with error messages, so when the query fails, we can do the check again to have a good error output.

Reply:
That's awesome!